server/user.go

```diff
@@ -1143,3 +1146,55 @@ func userUpdater() {
 Exit:
 	logs.Info.Println("users: shutdown")
 }
+
+// garbageCollectUsers runs every 'period' and deletes up to 'blockSize'
+// stale unvalidated user accounts which have been last updated at least
+// 'minAccountAgeDays' days.
+// Returns channel which can be used to stop the process.
+func garbageCollectUsers(period time.Duration, blockSize, minAccountAgeDays int, dryRun bool) chan<- bool {
+	// Unbuffered stop channel. Whomever stops the gc must wait for the process to finish.
+	stop := make(chan bool)
+	go func() {
+		// Add some randomness to the tick period to desynchronize runs on cluster nodes:
+		// 0.75 * period + rand(0, 0.5) * period.
+		period = (period >> 1) + (period >> 2) + time.Duration(rand.Intn(int(period>>1)))
+		gcTicker := time.Tick(period)
+		logs.Info.Printf("Starting account GC with period %s, block size %d and minimum account age %d days", period, blockSize, minAccountAgeDays)
+		staleAge := time.Hour * 24 * time.Duration(minAccountAgeDays)
+		for {
+			select {
+			case <-gcTicker:
+				if uids, authLvls, creds, err := store.Users.GetUnvalidatedUsers(time.Now().Add(-staleAge)); err == nil {
+					numUsers := len(uids)
+					if numUsers > blockSize {
+						numUsers = blockSize
+					}
+					var uidsToDelete []types.Uid
+					for i := 0; i < numUsers; i++ {
+						uid := uids[i]
+						authLvl := authLvls[i]
+						cred := strings.Split(creds[i], ",")
+						_, _, missing := stringSliceDelta(globals.authValidators[authLvl], cred)
+						if len(missing) > 0 {
+							uidsToDelete = append(uidsToDelete, uid)
+						}
+					}
+					logs.Info.Println("Stale account GC will delete the following uids", uidsToDelete)
+					if len(uidsToDelete) > 0 && !dryRun {
```

Comment:
I think the `logs.Info.Println("Stale account GC will delete the following uids")` from line 1182 should be moved under this if, otherwise it will add a useless constant log entry most of the time.

BTW, I'm not sure `dryRun` is needed for anything but debugging. Maybe remove it altogether?